### PR TITLE
IntelliJ/Java: Duplicate code -- Refactored as CLIReportUnexpectedExceptionHelper

### PR DESCRIPTION
**Pull request by jsoref**
This is for discussion.

Notes:
* Commits was probably created by IntelliJ for #4034
* In general, commits should not change program behavior at all

### Proposed changelog entries

* `Internal:` Internal Java code cleanup

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers
@Wadeck

### CODE REVIEW

**Comment by jsoref** [comment]
(This isn't ready, it's split so that it doesn't get lost)

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
No need in a new class, it can be put to CLICommand directly. If there is a justification for such new public helper class, it should be documented properly and put to a new package (jenkins.cli or so)

**Comment by fcojfernandez** [comment]
@oleg-nenashev requested changes on Jul and there has been no activity since then, so I'm proposing for close. @jsoref please, if you are still interested in this PR, let me know so I can remove the label

**Comment by jsoref** [comment]
@fcojfernandez: Sorry, I think when I first worked on this I couldn't figure out how to get the other package to access the higher class. Here we go.

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 1]
```diff
@@ -565,4 +560,12 @@ public static CLICommand getCurrent() {
         }
     }
 
+    static public void reportException(String name, Logger logger, PrintStream stderr, Throwable e) {
+        final String errorMsg = String.format("Unexpected exception occurred while performing %s command.",
+                name);
+        stderr.println();
+        stderr.println("ERROR: "+errorMsg);
```
Comment (zbynek) [inline core/src/main/java/hudson/cli/CLICommand.java]:
```suggestion
        stderr.println("ERROR: " + errorMsg);
```

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 2]
```diff
@@ -565,4 +560,12 @@ public static CLICommand getCurrent() {
...
+        logger.log(Level.WARNING,errorMsg,e);
```
Comment (zbynek) [inline core/src/main/java/hudson/cli/CLICommand.java]:
```suggestion
        logger.log(Level.WARNING, errorMsg, e);
```

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 3]
```diff
@@ -565,4 +560,12 @@ public static CLICommand getCurrent() {
...
+        Functions.printStackTrace(e,stderr);
```
Comment (zbynek) [inline core/src/main/java/hudson/cli/CLICommand.java]:
```suggestion
        Functions.printStackTrace(e, stderr);
```

**Review by zbynek — COMMENTED** [review]
Whitespace got removed in this PR,  IMHO it should be kept (consistency within file and with https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace )

**Review by fcojfernandez — APPROVED** [review]

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 4]
```diff
@@ -565,4 +560,12 @@ public static CLICommand getCurrent() {
         }
     }
 
+    static public void reportException(String name, Logger logger, PrintStream stderr, Throwable e) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/cli/CLICommand.java]: I would suggest to add Javadoc or to mark this API as restricted

Comment (jsoref) [reply core/src/main/java/hudson/cli/CLICommand.java]:
```suggestion
    @Restricted(NoExternalUse.class) 
    static public void reportException(String name, Logger logger, PrintStream stderr, Throwable e) {
```

Comment (MRamonLeon) [reply core/src/main/java/hudson/cli/CLICommand.java]: I think so jsoref, yes

**Comment by daniel-beck** [comment]
Not an API change if `@Restricted`.

**Review by varyvol — APPROVED** [review]
@jsoref Could you maybe resolve the merge conflict? This seems close to be ready once the conflict is solved.

**Comment by jsoref** [comment]
The conflicts are with #4641 which did something similar. I'll try to look at it over the weekend.

**Comment by daniel-beck** [comment]
The other PR mostly took care of other redundancy IIRC, these should nicely complement each other.

**Comment by varyvol** [comment]
@jsoref I've resolved the conflicts by making the function you introduced call the one introduced under #4641. Could you please confirm you are fine with it?

**Comment by jsoref** [comment]
@varyvol: thanks, lgtm

**Comment by jsoref** [comment]
@varyvol: now I remember why I didn't push that. I hit the static issues that the build has now. The functions in question have lost `this` and thus can't play nicely. I scratched my head and sighed.

**Comment by alecharp** [comment]
@jsoref do you think you can move this PR forward or should we close it? Thanks.

**Comment by alecharp** [comment]
Adding the `proposed-for-close label`, unless you have time to work on this @jsoref.

**Comment by jsoref** [comment]
I really can't figure out a good way to resolve this and I don't have the cycles to poke it.

**Pull request closed without merging**